examples/kubernetes/deployment.yaml

```diff
@@ -21,32 +21,39 @@ spec:
     spec:
       containers:
         - args:
-            - /usr/local/bin/synthetic-monitoring-agent --api-server-address=${API_SERVER} --api-token=${API_TOKEN} --verbose=true
-          command:
-            - sh
-            - -c
+            - --api-server-address=$(API_SERVER)
+            - --api-token=$(API_TOKEN)
+            - --verbose=true
+            # Uncomment to listen on all interfaces. This exposes metrics and debugging endpoints to the cluster, so it
+            # should only be done on trusted clusters or with network policies set in place.
+            # - --listen-address=0.0.0.0:4050
           env:
             - name: API_TOKEN
               valueFrom:
                 secretKeyRef:
                   key: api-token
                   name: sm-agent-1
-            #
             - name: API_SERVER
-              value: "PROBE_API_SERVER_URL"
+              valueFrom:
+                secretKeyRef:
+                  key: api-server
+                  name: sm-agent-1
           image: grafana/synthetic-monitoring-agent:latest
           name: agent
           ports:
             - containerPort: 4050
               name: http-metrics
-          readinessProbe:
```

Comment:
why we are commenting out readiness probe?

Reply:
By default the agent is not listening on all interfaces, which is required for liveness and readiness probes. See the new added comments that explain it in this PR.